socketio_block.py

```diff
@@ -36,63 +28,65 @@ class SocketIO(Retry, Block):
         version (enum): Which version of socketIO to use
 
     """
-    version = VersionProperty('1.1.0')
+    version = VersionProperty('2.0.0')
     host = StringProperty(title='SocketIo Host', default="127.0.0.1")
     port = IntProperty(title='Port', default=443)
-    room = StringProperty(title='SocketIo Room', default="default")
+    room = StringProperty(title='Socket.io Room', default="default")
     content = ExpressionProperty(
-        title='Content', default="{{json.dumps($to_dict(), default=str)}}",
+        title='Content', default="{{ json.dumps($to_dict(), default=str) }}",
         visible=False)
     listen = BoolProperty(title="Listen to SocketIo Room", default=False)
-    socketio_version = SelectProperty(
-        SocketIOVersion, title='Socket.IO Version', default=SocketIOVersion.v1)
+    connect_timeout = TimeDeltaProperty(
+        title="Connect timeout",
+        default={"seconds": 10},
+        visible=False)
 
     def __init__(self):
         super().__init__()
         self._sid = ""
-        self._heartbeat_job = None
         self._hb_interval = -1  # Heartbeat interval
         self._hb_timeout = -1  # Heartbeat timeout
         self._transports = ""  # Valid transports
         self._client = None
+        self._client_ready = False
+        # This bounded semaphore will ensure that only one thread can be
+        # connecting to the client at a time
+        self._connection_semaphore = BoundedSemaphore(1)
+        self._socket_url_protocol = "http"
         self._socket_url_base = ""
         self._stopping = False
-        self._reconnecting = False
 
     def configure(self, context):
         super().configure(context)
+        # For nio 2 compatibility
+        self.logger = self._logger
```

Comment:
Yeah, I left the `self.logger` references to make the merges into the nio2 branches easier. We say `self.logger` in nio 2 and unfortunately we're maintaining two versions of this block.